Source/inv.cpp

```diff
@@ -1808,18 +1808,21 @@ bool TryInvPut()
 	if (numitems >= MAXITEMS)
 		return false;
 
-	direction dir = GetDirection(plr[myplr].position.tile, { cursmx, cursmy });
-	if (CanPut(plr[myplr].position.tile.x + offset_x[dir], plr[myplr].position.tile.y + offset_y[dir])) {
+	Direction dir = GetDirection(plr[myplr].position.tile, { cursmx, cursmy });
+	Point position = plr[myplr].position.tile + dir;
+	if (CanPut(position.x, position.y)) {
 		return true;
 	}
 
-	direction dirLeft = left[dir];
-	if (CanPut(plr[myplr].position.tile.x + offset_x[dirLeft], plr[myplr].position.tile.y + offset_y[dirLeft])) {
+	Direction dirLeft = left[dir];
+	position = plr[myplr].position.tile + dirLeft;
+	if (CanPut(position.x, position.y)) {
 		return true;
 	}
 
-	direction dirRight = right[dir];
-	if (CanPut(plr[myplr].position.tile.x + offset_x[dirRight], plr[myplr].position.tile.y + offset_y[dirRight])) {
+	Direction dirRight = right[dir];
+	position = plr[myplr].position.tile + dirRight;

```

Comment:
```suggestion
	position = plr[myplr].position.tile + right[dir];
```

Also compared to the original this has come a long way :D
https://github.com/diasurgical/devilution/blob/f1b85a5cd3751363d248f7a2a460d527f69c2a32/Source/inv.cpp#L2378